server/server.go

```diff
@@ -412,7 +412,6 @@ func (s *Server) RegisterDaemon(instID flux.InstanceID, platform platform.Platfo
 	done := make(chan error)
 	s.messageBus.Subscribe(instID, s.instrumentPlatform(instID, platform), done)
 	err = <-done
-	close(done)
```

Comment:
This will get garbage collected when it goes out of scope, and there is nothing waiting on it so no reason to signal end-of-stream.